### PR TITLE
feat: Add macOS Qt builder script.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/TokTok/dockerfiles/187)

### CODE REVIEW

File: qtox/build_qt_macos.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+readonly SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+
+source "$SCRIPT_DIR/build_utils.sh"
+
+parse_arch --dep "qt" --supported "macos-arm64 macos-x86_64" "$@"
+
+QT_VERSION="6.8.1"
+
+brew install qt@6
```
Comment (nurupo) [inline qtox/build_qt_macos.sh:16]: Why is this needed?

Don't we build our own Qt6 from scratch here, why do we also need to install Qt6 from a package manager before we build ours?

**Review by nurupo — CHANGES_REQUESTED** [review]

File: qtox/build_qt_macos.sh (hunk 1 above)
Comment (iphydf) [reply qtox/build_qt_macos.sh:16]: Yes, Qt needs a Qt to bootstrap itself. After that, we don't need it to build additional modules.

Comment (iphydf) [reply qtox/build_qt_macos.sh]: Looks like I'm wrong. It's only needed for cross-compiling.

Comment (nurupo) [reply qtox/build_qt_macos.sh]: Yep, it's needed only for cross-compiling, and it's preferred if you build a native Qt off the same source (use native Qt of the same version) that you are cross-compiling, instead of using whatever Qt version is in the package manager. First you build a minimal native QtBase off the source, then you cross-compile a foreign QtBase with `-qt-host-path /native/qt/build/install/prefix` parameter.

**Review by nurupo — APPROVED** [review]

**Review by nurupo — APPROVED** [review]

**Review by nurupo — APPROVED** [review]
